src/proc_diskstats.c

```diff
@@ -69,21 +77,35 @@ struct disk *get_disk(unsigned long major, unsigned long minor) {
 		last->next = d;
 	}
 
+	// ------------------------------------------------------------------------
+	// find the type of the device
+
 	// find if it is a partition
-	// by reading /sys/dev/block/MAJOR:MINOR/partition
+	// by checking if /sys/dev/block/MAJOR:MINOR/partition is readable.
 	char buffer[FILENAME_MAX + 1];
-	snprintfz(buffer, FILENAME_MAX, path_find_block_device_partition, major, minor);
-
-	int fd = open(buffer, O_RDONLY, 0666);
-	if(likely(fd != -1)) {
-		// we opened it
-		int bytes = read(fd, buffer, FILENAME_MAX);
-		close(fd);
-
-		if(bytes > 0)
-			d->partition_id = strtoul(buffer, NULL, 10);
+	snprintfz(buffer, FILENAME_MAX, path_find_block_device, major, minor, "partition");
+	if(access(buffer, R_OK) == 0) {
+		d->type = DISK_TYPE_PARTITION;
+	} else {
+		// find if it is a container
+		// by checking if /sys/dev/block/MAJOR:MINOR/slaves has entries
+		snprintfz(buffer, FILENAME_MAX, path_find_block_device, major, minor, "slaves/");
+		DIR *dirp = opendir(buffer);	
+		if (dirp != NULL) {
+		struct dirent *dp;
+			while( (dp = readdir(dirp)) ) {
+				// . and .. are also files in empty folders.
+				if(strcmp(dp->d_name, ".") == 0 || strcmp(dp->d_name, "..") == 0) {
+					continue;
+				}
+				d->type = DISK_TYPE_CONTAINER;
+				// Stop the loop after we found one file.
+				break;
```

Comment:
> instead of d->type we could have d->slaves.

This won't work. Then we cannot distinguish between physical and virtual disks. This might not be relevant for default configuration but for the configuration options we like to provide.

Anyways counting `d->slaves` would solve my `LVM` setup, but I do not know if it is worthwhile to do it.
Imagine a `lvm` setup on one disk with various logical partitions.

There the logical volume group and the disk will be monitored and displayed. Because the group does only hold one disk the charts are quite the same.

To do the feature you suggest the right way we would need to find out all 1-1 relationships between "disks" and only display one of it. 

I can implement this if you want.
Personally I would not do such much effort on the default configurations. There are too many things we might miss and with the configurations options we have now most things can be configured with a few lines.

What do you think?